src/idaplugin/code_viewer.cpp

```diff
@@ -357,7 +357,8 @@ bool idaapi moveToPrevious()
 		}
 
 		decompInfo.decompiledFunction = fit->first;
-		qthread_create(showDecompiledCode, static_cast<void*>(&decompInfo));
+		ShowOutput show(&decompInfo);
```

Comment:
You are right about this being a remnant of an old design. Creating thread here (and other 3 places in this source and 1 place in `idaplugin.cpp`) is unnecessary.